src/erlfmt_cli.erl

```diff
@@ -196,51 +225,103 @@ read_stdin(Data) ->
-parse_opts([insert_pragma | _Rest], _Name, _Files, #config{pragma = require}) ->
+parse_opts([require_pragma | Rest], Files, Config) ->
+    parse_opts(Rest, Files, Config#config{pragma = require});
+parse_opts([insert_pragma | _Rest], _Files, #config{pragma = require}) ->
     {error, "Cannot use both --insert-pragma and --require-pragma options together."};
-parse_opts([insert_pragma | Rest], Name, Files, Config) ->
-    parse_opts(Rest, Name, Files, Config#config{pragma = insert});
-parse_opts([{files, NewFiles} | Rest], Name, Files0, Config) ->
-    parse_opts(Rest, Name, expand_files(NewFiles, Files0), Config);
-parse_opts([], _Name, [stdin], #config{out = Out}) when Out =/= standard_out, Out =/= check ->
+parse_opts([insert_pragma | Rest], Files, Config) ->
+    parse_opts(Rest, Files, Config#config{pragma = insert});
+parse_opts([{files, NewFiles} | Rest], Files0, Config) ->
+    parse_opts(Rest, expand_files(NewFiles, Files0), Config);
+parse_opts([], [stdin], #config{out = Out}) when
+    Out =/= standard_out, Out =/= undefined, Out =/= check
+->
     {error, "stdin mode can't be combined with write options"};
-parse_opts([], _Name, [stdin], Config) ->
+parse_opts([], [stdin], Config) ->
     {format, [stdin], Config};
-parse_opts([], _Name, [], _Config) ->
-    {error, "no files provided to format"};
-parse_opts([], _Name, Files, Config) ->
+parse_opts([], Files, Config) ->
     case lists:member(stdin, Files) of
         true -> {error, "stdin mode can't be combined with other files"};
         false -> {format, lists:reverse(Files), Config}
     end.
 
+-spec resolve_parsed(parsed(), parsed()) -> parsed().
+resolve_parsed(PreferParsed, DefaultParsed) ->
+    case {PreferParsed, DefaultParsed} of
+        {{error, _} = Error, _} ->
+            Error;
+        {_, {error, _} = Error} ->
+            Error;
+        {help, _} ->
+            help;
+        {_, help} ->
+            help;
+        {version, _} ->
+            version;
+        {_, version} ->
+            version;
+        {{format, PreferFiles, PreferConfig}, {format, DefaultFiles, DefaultConfig}} ->
+            {format, resolve_files(PreferFiles, DefaultFiles),
+                resolve_config(PreferConfig, DefaultConfig)}
+    end.
+
+resolve_files([], DefaultFiles) -> DefaultFiles;
+resolve_files(PreferFiles, _DefaultFiles) -> PreferFiles.
+
+resolve_config(
+    #config{verbose = PreferVerbose, width = PreferWidth, pragma = PreferPragma, out = PreferOut},
+    #config{
+        verbose = DefaultVerbose,
+        width = DefaultWidth,
+        pragma = DefaultPragma,
+        out = DefaultOut
+    }
+) ->
+    #config{
+        verbose = PreferVerbose orelse DefaultVerbose,
+        width = resolve_width(PreferWidth, DefaultWidth),
+        pragma = resolve_pragma(PreferPragma, DefaultPragma),
+        out = resolve_out(PreferOut, DefaultOut)
+    }.
+
+resolve_width(undefined, W) -> W;
+resolve_width(W, undefined) -> W;
+resolve_width(W, _) -> W.
```

Comment:
In this case the middle clause is not needed. The following would behave the same, bur perhaps it's clearer to have the extra clause, I'm not sure
```suggestion
resolve_width(undefined, W) -> W;
resolve_width(W, _) -> W.
```